client/my-sites/stats/stats-date-control/index.tsx

```diff
@@ -103,35 +103,30 @@ const StatsDateControl = ( { slug, queryParams, dateRange }: StatsDateControlPro
 	};
 
 	const getShortcutForRange = () => {
+		// Search the shortcut array for something matching the current date range.
+		// Returns shortcut or null;
 		const today = moment().format( 'YYYY-MM-DD' );
-		// Today
-		if ( today === dateRange.chartEnd && dateRange.daysInRange === 1 ) {
-			return shortcutList[ 0 ];
-		}
-		// Last 7 days
-		if ( today === dateRange.chartEnd && dateRange.daysInRange === 7 ) {
-			return shortcutList[ 2 ];
-		}
-		// Last 30 days
-		if ( today === dateRange.chartEnd && dateRange.daysInRange === 30 ) {
-			return shortcutList[ 3 ];
-		}
-		// Last year
-		if ( today === dateRange.chartEnd && dateRange.daysInRange === 365 ) {
-			return shortcutList[ 4 ];
-		}
 		const yesterday = moment().subtract( 1, 'days' ).format( 'YYYY-MM-DD' );
-		// Yesterday
-		if ( yesterday === dateRange.chartEnd && dateRange.daysInRange === 1 ) {
-			return shortcutList[ 1 ];
-		}
-		return null;
+		const shortcut = shortcutList.find( ( element ) => {
+			if (
+				yesterday === dateRange.chartEnd &&
+				dateRange.daysInRange === element.range + 1 &&
+				element.id === 'yesterday'
```

Comment:
I fixed it but the logic looks a bit confusing now..

Reply:
Were you seeing a bug with the logic here, @kangzj? I'm not sure I understand what the issue is. I'll do some testing with this again, just to be safe.